### ISSUE TITLE
Not supporting .NET Core

### ISSUE THREAD

**Issue by evoamac** [developerforce/Force.com-Toolkit-for-NET#196]
getting error "Package DeveloperForce.Force 1.3.0 is not compatible with netstandard1.5" when I run nuget.

**Comment by timcreatewell** [comment]
Hi there, I'm also keen to know if support for .net core is on the roadmap?

**Comment by evoamac** [comment]
Well, I end up downloaded the whole solution, and rewrote several
classes/methods. It works ok with .NET Core now

**Comment by timcreatewell** [comment]
Thanks @evoamac , I was thinking of doing the same and running the code through the Portability Analyzer (https://visualstudiogallery.msdn.microsoft.com/1177943e-cfb7-4822-a8a6-e56c7905292b) to see what would turn up.

**Comment by masaab** [comment]
I got it to work by adding net45 and removing netx45 in the project.json framework section.

**Comment by evoamac** [comment]
Is it targeted the newer netcore1.0 or the 5.0rc?

**Comment by masaab** [comment]
it is targeting netcore 1.0.
below is my project.json snippet.

 "dependencies": {
    "Microsoft.NETCore.App": {
      "version": "1.0.0-rc2-3002702",
      "type": "platform"
    },

**Comment by wadewegner** [comment]
Thanks for the info, @masaab. Is there anything I need to do here?

**Comment by wadewegner** [comment]
Incidentally, I ran the Portability Analyzer against the primary assembly.

![image](https://cloud.githubusercontent.com/assets/746259/17248481/ab15fd96-554f-11e6-86f8-172ed8caa59d.png)

**Comment by wadewegner** [comment]
And this:

![image](https://cloud.githubusercontent.com/assets/746259/17248502/c99055aa-554f-11e6-8f22-2f8c4d059506.png)

**Comment by masaab** [comment]
Looks good to me.

**Comment by mguinness** [comment]
Just tried package in a .NET Core project and got the following message:

Package DeveloperForce.Force 1.3.2 is not compatible with netcoreapp1.1

Is there any timeline for a core version?  Is there a working fork anywhere?

**Comment by masaab** [comment]
Try the following in your project.json. 
```
  "frameworks": {
    "netcoreapp1.0": {
      "imports": [
        "dotnet5.6",
        "netcoreapp1.0",
        "net45",
        "net451"
      ],
      "dependencies": {
      }
    }
  }
```

**Comment by mguinness** [comment]
@masaab - yeah that would work, but it would require the full framework.  I want a portable version that I can run on any platform like OSX/Linux.

**Comment by StingyJack** [comment]
I know microsoft has several tables that explain this stuff, but its a bit confusing when you get messages like `Package DeveloperForce.Force 1.3.2 supports: net (.NETFramework,Version=v0.0)`. I'm not sure that version "0" means "full framework required" or my VS/Tools/Dnx are broken. 

I used a similar config to get this to work. 
``` json
 "dependencies": {
    "DeveloperForce.Force": "1.3.2",
    "NETStandard.Library": "1.6.0"
  },

  "frameworks": {
    "netstandard1.6": {
      "imports": [ "dnxcore50", "net461", "net", "dotnet" ]
    }
    
    
```

**Comment by mguinness** [comment]
@StingyJack Altering the target framework forlder in the nuspec file should resolve this issue.

https://github.com/developerforce/Force.com-Toolkit-for-NET/pull/234

**Comment by cwharris** [comment]
@wadewegner The Portabilty report only tests whether or not the code itself is portable. It does not test the distribution method of the code. The code is 100% compatible, but the distribution method is 100% _incompatible_. Do you have plans to update the build targets to point at the most portable versions of .NET, so we can utilize this code using .NET Core on Mac and Unix? Microsoft is making a big push for these technologies, and it would be more than nice-to-have a compatible Force.com API client.

**Comment by ddrinka** [comment]
@wadewegner If you look at folks who have forked this project, 3 out of the most recent 4 have been to add .Net Core support.  As @cwharris said, things are really ramping up on the cross-platform front right now.  My use case is Amazon Web Services' Lambda--essentially Amazon's version of Heroku.  As of January Lambda supports .Net Core, and I'm migrating applications over.  This library would be great to drop into that eco-system.

I'll be working on my Lambda project over the next few weeks so if I can help get a netstandard1.3 build on nuget.org by backtesting for old .net versions or whatever, let me know.

**Comment by scastaldi** [comment]
@wadewegner great job with this plug-in, do you think it would be possible for you to make the SimpleConsole sample in dotnet core? me as well as many others in this thread are trying to use dotnet core from Linux. I will try @masaab solution but I don't feel confortable even if it works, referencing net45, the whole idea of using core is to keep it light.
In my case I'm new to core and linux, would be nice to have a working sample, so I don't have to bang my head in three places. Thanks in advance, even if I have to figure out how to do it in core the developerforce plug-in looks pretty solid

**Comment by mguinness** [comment]
@scastaldi if you reference net45 on Linux, you'll need Mono installed.  Otherwise using .NET Core requires libraries targeting .NET Standard and/or PCL, see [this answer on SO](http://stackoverflow.com/a/39721733/310601).  A possible solution is for the authors of this PCL library to target portable-net45+win8.

**Comment by scastaldi** [comment]
**# Good new guys!** 
I was able to run the SimpleConsole code **in my Ubuntu VM without installing Mono!!!**
When I get a little more time I will create a branch and give the sample code to @wadewegner  so we can have a supported sample in dotnet core. I want to spend a little more time to change the configuration from appconfig to  Secret Manager tool, so we don't expose sensitive information

For now this is the working project.json file
`
{
  "version": "1.0.0-*",
  "buildOptions": {
    "emitEntryPoint": true
  },
  "dependencies": {
    "Microsoft.NETCore.App": {
      "type": "platform",
      "version": "1.1.0"
    },
    "Microsoft.AspNetCore.Authentication.Cookies": "1.1.0",
    "Microsoft.AspNetCore.Diagnostics": "1.1.0",
    "Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore": "1.1.0",
    "Microsoft.AspNetCore.Identity.EntityFrameworkCore": "1.1.0",
    "DeveloperForce.Force": "1.4.0-beta",
    "ServicePointManagerExtensions": "1.0.2",
    "CoreCompat.System.Configuration": "4.2.3-r4",
    "System.Runtime.Serialization.Primitives": "4.3.0",
    "System.Runtime.Serialization.Formatters": "4.3.0",
    "System.Runtime.Serialization.Json": "4.3.0",
    "System.Runtime.Serialization.Xml": "4.3.0",
    "Newtonsoft.Json": "10.0.1"
  },
  "frameworks": {
    "netcoreapp1.1": {
      "imports": [
        "dotnet5.6",
        "netcoreapp1.0"
      ],
      "dependencies": {}
    }
  },
  "tooling": {
    "defaultNamespace": "force_console"
  }
}
`

**Comment by dodegaard** [comment]
@wadewegner I am working deep on .NET Core on things SF related and once the dust settles I will try to look at helping with a port.  Will be in touch. Let me know if we have any key people already on it.

**Comment by mguinness** [comment]
My recommendation would be for the new port to target .NET Standard 2.0 which is just around the corner (Q3 2017).  Although that limits use to .NET Core 2.0 it's a better long term solution and there is always this library to support earlier versions.

**Comment by mguinness** [comment]
@wadewegner Is there a roadmap for supporting .NET Core?  Now that .NET Standard 2.0 is out with it's expanded API surface it would seem like a good opportunity to release a version supporting both .NET Core 2.0 and .NET 4.6.1 framework.

**Comment by lazzDk** [comment]
Yes - It would be very much appreciated with .NET Core support

**Comment by shanekenyon87** [comment]
For .NET Core 2.0, this is a working csproj configuration (relevant sections):

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework> 
 <AssetTargetFallback>$(PackageTargetFallback);dotnet5.6;netcoreapp1.0;net45;net451</AssetTargetFallback>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="DeveloperForce.Force" Version="1.3.2" />
  </ItemGroup>

</Project>
```

**Comment by mguinness** [comment]
@shanekenyon87 Thanks for posting.  Frankly SF doesn't seem invested in ASP.NET Core and the last publish was Jan 2017.  Maybe a maintainer would like to give an update or roadmap for support?

**Comment by WazGettty** [comment]
@mguinness Agreed I've gone a way from this now as last update was a over a year ago. I've had to change to  .net core and include  big objects, so a roadmap would really help knowing if I'm left out on my own branch now.

**Comment by wadewegner** [comment]
I made significant updates and this SDK now targets .NET Standard 2 and .NET Core 2.

I haven't yet published update NuGet packages, but will soon. For now, you can get all the goodness by cloning the repo.

Thank you for your patience.
